internal/otelcollector/config/metric/agent/config_builder_test.go

```diff
@@ -248,6 +249,9 @@ func TestBuildAgentConfig(t *testing.T) {
 	})
 
 	t.Run("marshaling", func(t *testing.T) {
+		// To make the test passing in release branches
+		version.Version = "main"
+
```

Comment:
Can we inject the version via the Config instead of overriding the global variable?